include/exec/linux/io_uring_context.hpp

```diff
@@ -344,6 +344,12 @@ namespace exec {
         __wakeup_operation_.start();
       }
 
+      ~__context() noexcept {
+        if (!__pending_.empty()) {
+          run_until_empty();
+        }
+      }
```

Comment:
I know @kirkshoop has strong feelings about blocking in destructors, and good reasons for those feelings. This change makes me sus.

Reply:
Yes I know sorry. I shouldn't have pushed that. I will open a pr with a better solution. Currently an application terminates when the context is default constructed and run is not called at all. This is because the default constructor puts the wakeup op into the pending queue which has an assertion of  being not empty in its destructor.